service/k8sclient/k8sclient.go

```diff
@@ -1069,6 +1082,39 @@ func (c *K8sClient) CreatePSMDBCluster(ctx context.Context, params *PSMDBParams)
 	return c.kubeCtl.Apply(ctx, res)
 }
 
+func (c *K8sClient) generatePSMDBPasswords() (map[string][]byte, error) {
+	backup, err := generatePassword(passwordLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate backup password")
+	}
+	clusterAdmin, err := generatePassword(passwordLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate clusterAdmin password")
+	}
+	clusterMonitor, err := generatePassword(passwordLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate clusterMonitor password")
+	}
+	userAdmin, err := generatePassword(passwordLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate userAdmin password")
+	}
+
+	// secrets represents stringData part of
+	// https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/secrets.yaml.
+	secrets := map[string][]byte{
+		"MONGODB_BACKUP_USER":              []byte("backup"),
```

Comment:
similar here, I'd set only the password keys in the beginning, doing the same as for generateXtraDBPasswords, with small change -> adding *_USER keys befor returning. We may even reuse some code with couple of more changes. 
Something like
```

func generatePasswords(map[string][]byte) error {...}
```